src/with-component.js

```diff
@@ -1,13 +1,18 @@
 import { h, render } from 'preact';
 
-import { HTMLElement } from './util';
+import { HTMLElement, sym } from './util';
 import { withProps } from './with-props';
 import { withRender } from './with-render';
 import { withUnique } from './with-unique';
 
+const _preactDom = sym('_preactDom');
+
 export const withComponent = (Base = HTMLElement) => class extends withUnique(withRender(withProps(Base))) {
   rendererCallback (shadowRoot, renderCallback) {
-    render(renderCallback(), shadowRoot);
+    const dom = render(renderCallback(), shadowRoot, this[_preactDom]);
+    if (!this[_preactDom]) {
```

Comment:
Given you can get tens of millions of operations per second setting a property, I don't feel that it would matter too much. Happy to keep as is if you feel it's better. I think the Preact MO is just starting to rub off on me and I was trying to save a few bytes :)